packages/vite/src/node/utils.ts

```diff
@@ -78,12 +78,33 @@ const replaceSlashOrColonRE = /[/:]/g
 const replaceDotRE = /\./g
 const replaceNestedIdRE = /(\s*>\s*)/g
 const replaceHashRE = /#/g
-export const flattenId = (id: string): string =>
-  id
-    .replace(replaceSlashOrColonRE, '_')
-    .replace(replaceDotRE, '__')
-    .replace(replaceNestedIdRE, '___')
-    .replace(replaceHashRE, '____')
+export const flattenId = (id: string): string => {
+  const flatId = limitToCharacters(
+    id
+      .replace(replaceSlashOrColonRE, '_')
+      .replace(replaceDotRE, '__')
+      .replace(replaceNestedIdRE, '___')
+      .replace(replaceHashRE, '____'),
+  )
+  return flatId
+}
+
+const HASH_LENGTH = 10
+const MAX_FILE_LENGTH = 170
+
+export const limitToCharacters = (
```

Comment:
Lets rename to something more descriptive:
- `FLATTEN_ID_HASH_LENGTH`, let's make it 8 as we use in other places
- `FLATTEN_ID_MAX_FILE_LENGTH`
- We have a `getHash` function in util.ts already that we can use. You could pass a second parameter to it defaulting to 8, and use it here
- `limitFlattenIdLength()` and we don't need to export this function as it is only used by `flattenId`

Reply:
Thank you for the review! Working on it right now.